pytensor/tensor/basic.py

```diff
@@ -1888,6 +1888,25 @@ def _get_vector_length_MakeVector(op, var):
     return len(var.owner.inputs)
 
 
+@_vectorize_node.register
+def vectorize_make_vector(op: MakeVector, node, *batch_inputs):
+    # We vectorize make_vector as a join along the last axis of the broadcasted inputs
+    from pytensor.tensor.extra_ops import broadcast_arrays
+
+    # Check if we need to broadcast at all
+    bcast_pattern = batch_inputs[0].type.broadcastable
+    if not all(
+        batch_input.type.broadcastable == bcast_pattern for batch_input in batch_inputs
+    ):
+        batch_inputs = broadcast_arrays(*batch_inputs)
+
+    # Join along the last axis
+    new_out = join(
+        -1, *[expand_dims(batch_inputs, axis=-1) for batch_inputs in batch_inputs]
+    )
+    return new_out.owner
```

Comment:
why `.owner` here?

Reply:
`vectorize_node` returns a node. No idea why, could probably have made it like rewrites and return the output variables

Reply:
I guess it's right there in the name eh?

Reply:
It's silly and potentially too restrictive, opened an issue: https://github.com/pymc-devs/pytensor/issues/902